docs/reference/numbers.tex

```diff
@@ -0,0 +1,601 @@
+as summarized in \tref{nums.summary}.
+
+\begin{libsumtab}{Numbers library summary}{nums.summary}
+\ref{num.traits}        & Traits                                 & \tcode{<mp_units/numbers.h>} \\
+\ref{num.concepts}      & Concepts                               & \\
+\end{libsumtab}
+
+\Sec1[nums.syn]{Header \tcode{<mp_units/numbers.h>} synopsis}
+
+\indexheader{mp_units/numbers.h}%
+\begin{codeblock}
+namespace mp_units {
+
+// \ref{num.traits}, traits
+
+// \ref{num.opt.ins}, \cname{number} opt-ins
+template<typename T>
+struct is_number;
+template<typename T>
+struct is_complex_number;
+
+template<typename T>
+constexpr bool @\libglobal{is_number_v}@ = is_number<T>::value;
+template<typename T>
+constexpr bool @\libglobal{is_complex_number_v}@ = is_complex_number<T>::value;
+
+// \ref{num.ids}, identities
+template<typename T>
+struct number_zero;
+template<typename T>
+struct number_one;
+
+template<typename T>
+constexpr T @\libglobal{number_zero_v}@ = number_zero<T>::value;
+template<typename T>
+constexpr T @\libglobal{number_one_v}@ = number_one<T>::value;
+
+// \ref{num.assoc.types}, associated types
+template<typename T>
+struct number_scalar;
+
+template<typename T>
+using number_difference_t = decltype(std::declval<const T>() - std::declval<const T>());
+template<typename T>
+using @\libglobal{number_scalar_t}@ = number_scalar<T>::type;
+
+// \ref{num.concepts}, concepts
+template<typename T>
+concept number = @\seebelow@;
+template<typename T>
+concept ordered_number = @\seebelow@;
+template<typename T>
+concept number_line = @\seebelow@;
+template<typename T, typename U>
+concept modulo_with = @\seebelow@;
+template<typename T, typename U>
+concept compound_modulo_with = @\seebelow@;
+template<typename T, typename U>
+concept modulus_for = @\seebelow@;
+template<typename T, typename U>
+concept compound_modulus_for = @\seebelow@;
+template<typename T>
+concept negative = @\seebelow@;
+template<typename T>
+concept set_with_inverse = @\seebelow@;
+template<typename T, typename U>
+concept point_space_for = @\seebelow@;
+template<typename T, typename U>
+concept compound_point_space_for = @\seebelow@;
+template<typename T>
+concept point_space = @\seebelow@;
+template<typename T, typename U>
+concept vector_space_for = @\seebelow@;
+template<typename T, typename U>
+concept compound_vector_space_for = @\seebelow@;
+template<typename T, typename U>
+concept scalar_for = @\seebelow@;
+template<typename T, typename U>
+concept field_for = @\seebelow@;
+template<typename T, typename U>
+concept compound_scalar_for = @\seebelow@;
+template<typename T, typename U>
+concept compound_field_for = @\seebelow@;
+template<typename T>
+concept vector_space = @\seebelow@;
+template<typename T>
+concept f_vector_space = @\seebelow@;
+template<typename T>
+concept field_number = @\seebelow@;
+template<typename T>
+concept field_number_line = @\seebelow@;
+template<typename T>
+concept scalar_number = @\seebelow@;
```

Comment:
Thank you for your support, Mateusz!
I have formed an idea on how to more formally present this to SG6
(instead of just [dropping the PDF](https://github.com/mpusz/mp-units/issues/29#issuecomment-1712552324) on their channel in the cpplang slack).
When I have something more concrete written down,
I'll share its link to <https://github.com/JohelEGP/proposals> here.

Reply:
> how to more formally present this to SG6

The best would be to present it as a paper in Kona 😉 

In case you will have it ready earlier, I can send it to a few people for feedback so you can consider some changes before the final deadline.